### PR TITLE
Add Mesh ConvexDecompositionSettings wrapper

### LINKED ISSUE #67801: create_multiple_collision_shapes() does not do convex decomposition
### Godot version

4.0 dev( 4.1 beta, 4.2 beta, 4.3 beta)

### System information

OS Version : MacOS Monterrey, Hardware : Mac OS Silicon 

### Issue description

When trying to automatically create multiple convex collision shapes from a concave MeshInstance3D, only one convex collision shape is created instead of multiple. This happens in the editor and when calling create_multiple_convex_collisions() in code on a MeshInstance3D.

### Steps to reproduce

In editor: 

1. Create a new 3D scene with Node3D as root.
2. Add RigidBody3D as child of scene root
3. Add MeshInstance3D as child of RigidBody3D
4. Assign TorusMesh to the mesh of the MeshInstance3D
5. Select the MeshInstance3D in the scene tree
6. Click the Mesh icon next to the view icon in the editor
7. Click Create Multiple Convex Collision Siblings
8. Observe that only one convex collision shape is added as a child to the RigidBody3D

In code:

1. Create a torus mesh as the mesh of a MeshInstance3D that is the child of a rigid body
2. call create_multiple_convex_collisions() on the mesh instance
3. reparent the collision shapes to be the siblings of the MeshInstance3D
4. put a rigid body with a collision shape that fits in the hole of the torus, and notice that it does not fall through, because the only one convex collision shape is being created for the entire mesh even though it is concave.

### Minimal reproduction project

[ExampleOfBug.zip](https://github.com/godotengine/godot/files/9846734/ExampleOfBug.zip)

### PR DESCRIPTION
**Pull request by smix8**
Adds wrapper `MeshConvexDecompositionSettings` to control the many parameters for Mesh `ConvexDecomposition` operations with the VHACD library. Changes all uses of the old `struct` Mesh::ConvexDecompositionSettings to the new ref.

Previously Godot had no option for script users to change the parameters. Some inappropiated values for certain meshes and operations were more or less hardcoded defaults like limiting the entire hull count to a single hull or forcing maximum concavity.

Fixes https://github.com/godotengine/godot/issues/67801 and similar situations where different parameters are required to get a usable result.

``` gdscript
# those are just random test parameters as an example
# not suited for all meshes and especially not for runtime

var settings = MeshConvexDecompositionSettings.new()

settings.convexhull_approximation = true
settings.max_convex_hulls = 1024
settings.convexhull_downsampling = 4
settings.plane_downsampling = 4
settings.max_num_vertices_per_convex_hull = 64
settings.max_concavity = 0.001
settings.min_volume_per_convex_hull = 0.0001
settings.project_hull_vertices = true
settings.resolution = 1000000
settings.revolution_axes_clipping_bias = 0.05
settings.symmetry_planes_clipping_bias = 0.05
settings.normalize_mesh = false

MeshInstance3D.create_multiple_convex_collisions(settings)
```
The only function currently exposed to scripting users that is working with the new parameters is `MeshInstance3D.create_multiple_convex_collisions(settings)`.

Did not change the default values but some of them should be changed at some point when tested properly e.g. the max hull count. The Editor MeshInstance3D "Mesh" Button could also probably benefit from a GUI that allows to change settings but that is stuff the Editor GUI ppl can think about ;)

### CODE REVIEW

**Review by fire — APPROVED** [review]
I agree with the changes, these are mostly documentation and exposing configs.

I am unable to test as we have a few prs to get done.

**Comment by CornbreadWithMarmalade** [comment]
Seems like this should be added to godot 4.0 instead of 4.1

**Comment by Calinou** [comment]
> Seems like this should be added to godot 4.0 instead of 4.1

4.0 is in feature freze, so any new features are targeting 4.1 or later from now on.

File: doc/classes/MeshConvexDecompositionSettings.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	<description>
+		Parameters to be used with a [Mesh] convex decomposition operation.
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="convexhull_approximation" type="bool" setter="set_convexhull_approximation" getter="get_convexhull_approximation" default="true">
+			If enabled uses approximation for computing convex hulls.
+		</member>
+		<member name="convexhull_downsampling" type="int" setter="set_convexhull_downsampling" getter="get_convexhull_downsampling" default="4">
```
Comment (YuriSizov) [inline doc/classes/MeshConvexDecompositionSettings.xml:15]: "convexhull" should be 2 words, me thinks. Also please use `[code]` around numeric literals in the descriptions.

File: scene/resources/mesh.cpp [hunk 2]
```diff
@@ -625,6 +627,168 @@ Ref<Resource> Mesh::create_placeholder() const {
 	return placeholder;
 }
 
+void MeshConvexDecompositionSettings::_bind_methods() {
```
Comment (YuriSizov) [inline scene/resources/mesh.cpp:630]: We usually put this method at the end, before the constructor if there is any.

File: scene/resources/mesh.h [hunk 3]
```diff
@@ -225,6 +195,89 @@ class Mesh : public Resource {
...
+	static void _bind_methods();
+
+public:
+	enum Mode : int {
+		CONVEX_DECOMPOSITION_MODE_VOXEL = 0,
+		CONVEX_DECOMPOSITION_MODE_TETRAHEDRON = 1
+	};
+
+private:
+	Mode mode = CONVEX_DECOMPOSITION_MODE_VOXEL;
```
Comment (YuriSizov) [inline scene/resources/mesh.h:236]: We usually put this before the methods, somewhere next to other properties.

**Review by YuriSizov — APPROVED** [review]
Seems like a nice improvement, aside from the suggestions listed above.

File: doc/classes/MeshConvexDecompositionSettings.xml (hunk 1 above)
Comment (smix8) [reply doc/classes/MeshConvexDecompositionSettings.xml:15]: I have no opinion if this should be 1 or 2 words, it is a 1:1 copy on how the variables were named before. I did a quick search and there are cases throughout the engine for both namings so if we want to unify them we properly should do all in one sweep.

Comment (YuriSizov) [reply doc/classes/MeshConvexDecompositionSettings.xml:15]: From what I see, in this same exposed API you mostly have it as two words. So IMO this particular case should be addressed in this PR. If there are any other cases in public names that don't follow that convention, we probably can't do much about it at this point (backwards compatibility and all). But this API here is new, and is inconsistent between itself.

**Comment by YuriSizov** [comment]
Thanks!

**Comment by EzraT** [comment]
Thank you!

**Pull request merged**